src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -0,0 +1,392 @@
+        }
+        if (offset < 0) {
+            throw new GATKException("Illegal offset < 0 UnifiedReadBackedPileup");
+        }
+
+        final List<PileupElement> pileup = new LinkedList<>();
+        for (final GATKRead read : reads) {
+            pileup.add(PileupElement.createPileupForReadAndOffset(read, offset));
+        }
+
+        return pileup;
+    }
+
+    /**
+     * Make a new pileup consisting of elements of this pileup that satisfy the predicate.
+     */
+    public ReadPileup makeFilteredPileup(final Predicate<PileupElement> filter){
+        return new ReadPileup(loc, pileupElements.stream().filter(filter).collect(Collectors.toList()));
+    }
+
+    /**
+     * Make a new pileup consisting of only reads on the negative strand.
+     */
+    public ReadPileup getPositiveStrandPileup() {
+        return makeFilteredPileup(p -> !p.getRead().isReverseStrand());
+    }
+
+    /**
+     * Make a new pileup consisting of only reads on the negative strand.
+     */
+    public ReadPileup getNegativeStrandPileup() {
+        return makeFilteredPileup(p -> p.getRead().isReverseStrand());
+    }
+
+    /**
+     * Make a new pileup that contains of only bases with quality >= minBaseQ, coming from
+     * reads with mapping qualities >= minMapQ.
+     */
+    public ReadPileup getBaseAndMappingFilteredPileup(final int minBaseQ, final int minMapQ) {
+        return makeFilteredPileup(p -> p.getRead().getMappingQuality() >= minMapQ && (p.isDeletion() || p.getQual() >= minBaseQ));
+    }
+
+    /**
+     * Make a new pileup that contains of only bases with quality >= minBaseQ.
+     */
+    public ReadPileup getBaseFilteredPileup(final int minBaseQ) {
+        return makeFilteredPileup(p -> (p.isDeletion() || p.getQual() >= minBaseQ));
+    }
+
+    /**
+     * Make a new pileup that contains of only bases coming from reads with mapping quality >= minMapQ.
+     */
+    public ReadPileup getMappingFilteredPileup(final int minMapQ) {
+        return makeFilteredPileup(p -> p.getRead().getMappingQuality() >= minMapQ);
+    }
+
+    /**
+     * Makes a new pileup by subsetting reads from a given read group.
+     * Returns null if no reads are from the given read group.
+     * @param targetReadGroupId Identifier for the read group.
+     * @return A read-backed pileup containing only the reads in the given read group.
+     */
+    public ReadPileup getPileupForReadGroup(final String targetReadGroupId) {
+        final ReadPileup pu = makeFilteredPileup(p -> {
+            final GATKRead read = p.getRead();
+            if (targetReadGroupId != null) {
+                if (read.getReadGroup() != null && targetReadGroupId.equals(read.getReadGroup())) {
+                    return true;
+                }
+            } else {
+                if (read.getReadGroup() == null || read.getReadGroup() == null) {
+                    return true;
+                }
+            }
+            return false;
```

Comment:
i replaced it with `Objects.equals(targetReadGroupId, p.getRead().getReadGroup())`